### PR TITLE
Emmanuel/dao/user interface tests

### PR DESCRIPTION
**Pull request by emmanuel099**
* Added User DTO
* Added User DAO Interface + Tests
* Added empty User DAO Impl
* Changed datatype of user id to String (VARCHAR(25) in database)

### CODE REVIEW

File: src/main/resources/create.sql [hunk 1]
```diff
@@ -1,17 +1,17 @@
 CREATE TABLE IF NOT EXISTS RestaurantUser (
-    ID IDENTITY,
+    ID VARCHAR(25),
```
Comment (PhilippParis) [inline src/main/resources/create.sql]: use "SVNR" as ID?

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+        User user = new User();
+        user.setIdentity("jaunty");
+        user.setRole("tester");
+
+        // WHEN
+        userDAO.create(user);
+    }
+
+    @Test(expected = ValidationException.class)
+    public void testCreate_addingObjectWithoutRoleShouldFail() throws DAOException, ValidationException {
```
Comment (PhilippParis) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java]: identity must be set before create()?

documentation of the create method:
"sets the identity parameter of user, if the data was stored successfully"

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+        // GIVEN
+        User matcher = new User();
+        matcher.setIdentity("zzzzzzzzzzzzzz");
+        assertTrue(userDAO.find(matcher).isEmpty());
+
+        // WHEN
+        List<User> result = userDAO.find(matcher);
+
+        // THEN
+        assertTrue(result.isEmpty());
```
Comment (PhilippParis) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java]: assertTrue(userDAO.find(matcher).isEmpty());
==
List<User> result = userDAO.find(matcher);
assertTrue(result.isEmpty());

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java (hunk 2 above)
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java]: Good catch!
The documentation is outdated

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java (hunk 3 above)
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestUserDAO.java]: fixed, search user identities by using buildRandomString

**Pull request merged**